%%% sunny-sweep-1 wua96xrj %%%
Traced<ShapedArray(float32[3])>with<DynamicJaxprTrace(level=1/0)>
Traceback (most recent call last):
  File "C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\train_loops.py", line 129, in wandb_train_td3
    td3.buffer.add(obs, action, next_obs, reward, int(done))
                                                  ^^^^^^^^^
  File "C:\Users\31614\anaconda3\Lib\site-packages\jax\_src\core.py", line 785, in __int__
    return self.aval._int(self)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\31614\anaconda3\Lib\site-packages\jax\_src\core.py", line 1553, in error
    raise ConcretizationTypeError(arg, fname_context)
jax.errors.ConcretizationTypeError: Abstract tracer value encountered where concrete value is expected: traced array with shape bool[]
The problem arose with the `int` function. If trying to convert the data type of a value, try using `x.astype(int)` or `jnp.array(x, int)` instead.
The error occurred while tracing the function wandb_train_td3 at C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\train_loops.py:82 for jit. This value became a tracer due to JAX operations on these lines:

  operation a[35m:i32[][39m = convert_element_type[new_dtype=int32 weak_type=False] b
    from line C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\train_loops.py:84:10 (wandb_train_td3)

  operation a[35m:i32[][39m = convert_element_type[new_dtype=int32 weak_type=False] b
    from line C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\model.py:21:26 (TD3.__init__)

  operation a[35m:u32[][39m = convert_element_type[new_dtype=uint32 weak_type=False] b
    from line C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\model.py:21:26 (TD3.__init__)

  operation a[35m:i32[][39m = convert_element_type[new_dtype=int32 weak_type=False] b
    from line C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\model.py:22:19 (TD3.__init__)

  operation a[35m:u32[][39m = convert_element_type[new_dtype=uint32 weak_type=False] b
    from line C:\Users\31614\Documents\AI_course_stuff\RLForaging Thesis\RLForagingAgents\RL-models\TD3\parameters.py:6:18 (CriticTD3.__init__)

(Additional originating lines are not shown.)

See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.ConcretizationTypeError
